ethash/src/compute.rs

```diff
@@ -137,11 +137,11 @@ fn hash_compute(light: &Light, full_size: usize, header_hash: &H256, nonce: u64)
 		($n:expr, $value:expr) => {{
 			// We use explicit lifetimes to ensure that val's borrow is invalidated until the
 			// transmuted val dies.
-			unsafe fn make_const_array<'a, T, U>(val: &'a mut [T]) -> &'a mut [U; $n] {
+			unsafe fn make_const_array<'a, T, U>(val: &mut [T]) -> &mut [U; $n] {
```

Comment:
Drop `unsafe`?

Reply:
`&mut *(val.as_mut_ptr() as *mut [U; $n])` is also unsafe because `deref` of raw pointer so I would like to keep the function as `unsafe`! Thoughts?

Reply:
Yeah won't compile without `unsafe` ignore my comment.

Reply:
looks like the explicit lifetimes have been dropped here. should be `&'a mut [T] -> &'a mut [U; $n]` right?

Reply:
@rphmeier Explicit `lifetimes` here are not needed here, so I removed them but forgot `a` in the signature!

I can revert this if you want explicit lifetime annotations!